src/librbd/crypto/openssl/OpenSSLDataCryptor.cc

```diff
@@ -0,0 +1,84 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "librbd/crypto/openssl/OpenSSLDataCryptor.h"
+#include <string.h>
+#include "include/ceph_assert.h"
+
+namespace librbd {
+namespace crypto {
+namespace openssl {
+
+OpenSSLDataCryptor::OpenSSLDataCryptor(const char* cipher_name,
+                                       const unsigned char* key,
+                                       uint16_t key_length) {
+  ceph_assert(cipher_name != nullptr);
+  m_cipher = EVP_get_cipherbyname(cipher_name);
+  ceph_assert(m_cipher != nullptr);
```

Comment:
Nit: might be nicer to delay the initialization until an `init()` method that can return a status code (instead of raising an assertion) here and below. That might mean you would rename the current `init` / `update` methods to `init_context` / `update_context`

Reply:
Should I declare an error code for each possible error? If so, should I put it in DataCryptor.h?

Reply:
You can use `ldout(CephContext*, level) << "error description" << dendl` to log the error and then just return an appropriate (negative) POSIX error code.